.github/workflows/checks.yml

```diff
@@ -0,0 +1,63 @@
+name: CI
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/checks.yaml'
+      - '**.go'
+
+permissions:
+  # Permission for checking out code
+  contents: read
+
+jobs:
+  build:
+    name: Check & Build Provider
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - run: chmod +x ./scripts/gofmtcheck.sh
+      - name: gofmt Check
+        run: ./scripts/gofmtcheck.sh
+      # - name: Generate provider code
+      #   run: go generate
+      # - name: Check generated code for diffs
+      #   run: git diff --exit-code
+      - name: Build
+        run: go build -v
+  acceptance:
+    name: Acceptance Tests
+    needs: build
+    runs-on: ubuntu-latest
+    concurrency:
+        group: tf-aci-ci-test-${{ matrix.apic_host.name }}
+        cancel-in-progress: false
+    strategy:
+      fail-fast: false
+      matrix:
+        apic_host:
+          - name: v5.2
+            url: 'https://173.36.219.83/'

```

Comment:
Once this is merged, I will create a different PR to update the hosts and add CI user/password.